src/OmniSharp.DotNet/Cache/ProjectStatesCache.cs

```diff
@@ -128,49 +190,34 @@ private ProjectEntry GetOrAddEntry(string projectDirectory)
             }
         }
 
-        private class ProjectEntry
+        private ProjectEntry GetOrAddEntry(string projectDirectory, out bool added)
         {
-            private readonly Dictionary<NuGetFramework, ProjectState> _states
-                       = new Dictionary<NuGetFramework, ProjectState>();
-
-            public ProjectEntry(string projectDirectory)
+            added = false;
+            ProjectEntry result;
+            if (_projects.TryGetValue(projectDirectory, out result))
             {
-                ProjectDirectory = projectDirectory;
-            }
-
-            public string ProjectDirectory { get; }
-
-            public IEnumerable<NuGetFramework> Frameworks => _states.Keys;
-
-            public IEnumerable<ProjectState> ProjectStates => _states.Values;
 
-            public ProjectState Get(NuGetFramework framework)
-            {
-                ProjectState result;
-                if (_states.TryGetValue(framework, out result))
-                {
-                    return result;
-                }
-                else
-                {
-                    return null;
-                }
+                return result;
             }
-
-            public void Set(ProjectState state)
+            else
             {
-                _states[state.ProjectContext.TargetFramework] = state;
-            }
+                result = new ProjectEntry(projectDirectory);
+                _projects[projectDirectory] = result;
+                added = true;
 
-            public bool Remove(NuGetFramework framework)
-            {
-                return _states.Remove(framework);
+                return result;
             }
+        }
 
-            public override string ToString()
-            {
-                return $"ProjectEntry {ProjectDirectory}, {_states.Count} states";
-            }
+        private void EmitProject(string eventType, DotNetProjectInformation information) {
+            _emitter.Emit(
+                eventType,
+                new ProjectInformationResponse()
+                {
+                    { "DotNetProject", information },
+                    // the key is hard coded in VSCode
+                    { "DnxProject", information }
```

Comment:
I'm not sure why it's necessary at this point. VS Code has DnxProject while we're targeting an older OmniSharp, but I'm not sure why we need this here.